src/models/StyledComponent.js

```diff
@@ -25,23 +25,13 @@ const createStyledComponent = (target: Target, rules: RuleSet) => {
     static rules: RuleSet
     static target: Target
 
-    getChildContext() {
-      return { theme: this.theme }
-    }
-
     componentWillMount() {
       this.componentWillReceiveProps(this.props, this.context)
     }
 
     componentWillReceiveProps(newProps: Object, newContext: ?any) {
       // Always pass down a theme, even if it's empty
-      this.theme = (newContext && newContext.theme) || {}
-      // Local copy for this instance with an update() method
-      const theme = Object.assign({}, this.theme, {
-        update: values => {
-          this.theme = Object.assign({}, this.theme, values)
-        },
-      })
+      const theme = (newContext && newContext.theme) || {}
```

Comment:
Probably, but as I noted:

>it's best to avoid using it so you don't add too many wrappers.

It's bad for perf if every styled div is actually two composite React components. We need to keep nesting as flat as possible. So it's better to reimplement that logic by hand (it's pretty simple).